### ISSUE TITLE
Allow administrators to archive promotions

### ISSUE THREAD

**Issue by loevgaard** [Sylius/Sylius#13475]
**Describe the proposed solution**
![image](https://user-images.githubusercontent.com/2412177/149148178-2682f2f7-a28a-4c41-9b1c-9181d00ce010.png)

**Additional context**
I have a customer who has hundreds of promotions and because they either have expiry dates or usage limits, they are not relevant anymore and this feature would make a lot of sense.

**Comment by vvasiloi** [comment]
There's should be a way to still view the archived promotions if necessary. Maybe with a grid filter.

**Comment by CoderMaggie** [comment]
Oh, so many entities could have this feature added 😓  

One day guys, one day 😅

**Comment by vvasiloi** [comment]
@CoderMaggie can you make a list with those entities in a separate issue?

**Comment by loevgaard** [comment]
Products and variants would be on that list :D

**Comment by vvasiloi** [comment]
And maybe locales... since there's currently no way to delete them from admin panel and users keep installing the default fixtures in production environments, having a lot of locales they don't need...

**Comment by loevgaard** [comment]
You are right that it's a mess to actually remove a locale. A lot of manual sql queries :D Not sure it would be just as easy as with promos and products to fulfil.

**Comment by TheMilek** [comment]
Hi, just letting you know that We are on it and just added it to our backlog 😄

**Comment by loevgaard** [comment]
Cool :D

**Comment by senghe** [comment]
Hi all! :) Just wanted to tell you, that I've already pushed some code to allow Sylius to archive promotions and catalog promotions :tada: I see now there is broken PHPStan etc. but I hope it'll be merged soon :crossed_fingers:

**Comment by senghe** [comment]
The PR is ready for review :vulcan_salute:
